code/game/machinery/doors/firedoor.dm

```diff
@@ -130,15 +134,40 @@
 		log_game("[key_name(user)] [welded ? "welded":"unwelded"] firedoor [src] with [W] at [AREACOORD(src)]")
 		update_appearance()
 
-/obj/machinery/door/firedoor/try_to_crowbar(obj/item/I, mob/user)
+/// We check for adjacency when using the primary attack.
+/obj/machinery/door/firedoor/try_to_crowbar(obj/item/acting_object, mob/user)
 	if(welded || operating)
 		return
 
 	if(density)
+		being_held_open = TRUE
 		open()
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/handle_held_open_adjacency)
+		RegisterSignal(user, COMSIG_LIVING_SET_BODY_POSITION, .proc/handle_held_open_adjacency)
```

Comment:
This should also check if the mob is deleted for whatever reason.

Reply:
Alright, i changed it to check for the user and if it's being qdeleted. 
Is this sufficient?

Reply:
Wait, the define itself handles that, be right back.